python/ray/data/_internal/stats.py

```diff
@@ -275,137 +332,302 @@ def summary_string(
+        if memory_stats:
             out += indent
-            memory_stats = [
-                round(e.max_rss_bytes / (1024 * 1024), 2) for e in exec_stats
-            ]
             out += "* Peak heap memory usage (MiB): {} min, {} max, {} mean\n".format(
-                min(memory_stats),
-                max(memory_stats),
-                int(np.mean(memory_stats)),
+                memory_stats["min"],
+                memory_stats["max"],
+                memory_stats["mean"],
             )
 
-        output_num_rows = [m.num_rows for m in blocks if m.num_rows is not None]
-        if output_num_rows:
+        output_num_rows_stats = getattr(self, StageStatsSummary.OUTPUT_NUM_ROWS, {})
+        if output_num_rows_stats:
             out += indent
             out += "* Output num rows: {} min, {} max, {} mean, {} total\n".format(
-                min(output_num_rows),
-                max(output_num_rows),
-                int(np.mean(output_num_rows)),
-                sum(output_num_rows),
+                output_num_rows_stats["min"],
+                output_num_rows_stats["max"],
+                output_num_rows_stats["mean"],
+                output_num_rows_stats["sum"],
             )
 
-        output_size_bytes = [m.size_bytes for m in blocks if m.size_bytes is not None]
-        if output_size_bytes:
+        output_size_bytes_stats = getattr(self, StageStatsSummary.OUTPUT_SIZE_BYTES, {})
+        if output_size_bytes_stats:
             out += indent
             out += "* Output size bytes: {} min, {} max, {} mean, {} total\n".format(
-                min(output_size_bytes),
-                max(output_size_bytes),
-                int(np.mean(output_size_bytes)),
-                sum(output_size_bytes),
+                output_size_bytes_stats["min"],
+                output_size_bytes_stats["max"],
+                output_size_bytes_stats["mean"],
+                output_size_bytes_stats["sum"],
             )
 
-        if exec_stats:
-            node_counts = collections.defaultdict(int)
-            for s in exec_stats:
-                node_counts[s.node_id] += 1
+        node_count_stats = getattr(self, StageStatsSummary.NODE_COUNT, {})
+        if node_count_stats:
             out += indent
             out += "* Tasks per node: {} min, {} max, {} mean; {} nodes used\n".format(
-                min(node_counts.values()),
-                max(node_counts.values()),
-                int(np.mean(list(node_counts.values()))),
-                len(node_counts),
+                node_count_stats["min"],
+                node_count_stats["max"],
+                node_count_stats["mean"],
+                node_count_stats["count"],
             )
+        return out
 
-        if self.extra_metrics:
-            out += indent
-            out += "* Extra metrics: " + str(self.extra_metrics) + "\n"
 
+@dataclass
+class IterStatsSummary:
+    wait_time: Timer
```

Comment:
Can you add a brief documentation about the meaning of each stats?
Maybe for StageStatsSummary as well.